osu.Game.Rulesets.Osu/Difficulty/Evaluators/AimEvaluator.cs

```diff
@@ -33,8 +39,46 @@ public static double EvaluateDifficultyOf(DifficultyHitObject current, bool with
             var osuLastObj = (OsuDifficultyHitObject)current.Previous(0);
             var osuLastLastObj = (OsuDifficultyHitObject)current.Previous(1);
 
+            // Doubletap detection stuff
+
+            // Calculate how much things are overlapping. Low overlapping circles are very hard in being doubletappable
+            double overlapnessCurr = Math.Clamp(osuCurrObj.LazyJumpDistance / OsuDifficultyHitObject.NORMALISED_RADIUS / 2, 0, 1);
+            overlapnessCurr = CosSigmoid(overlapnessCurr);
+            overlapnessCurr *= overlapnessCurr;
+
+            double antiOverlapnessLast = 1 - Math.Clamp(osuLastObj.LazyJumpDistance / OsuDifficultyHitObject.NORMALISED_RADIUS / 2, 0, 1);
```

Comment:
Not sure if its just for me, but I think this variable name reads terribly

Reply:
antiOverlapness sounds very weird but idk how i can name it better without making it super long
antiOverlapness = 1 - overlapness
but i can leave "1 - x" operation in the end cuz it's transformed slightly after